File: software/examples/javascript/ExampleCallback.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'jQs'; // Change to your UID

var ipcon = new Tinkerforge.IPConnection(); // Create IP connection
var mt = new Tinkerforge.BrickletMultiTouch(UID, ipcon); // Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);
    }
); // Connect to brickd
// Don't use device before ipcon is connected

// Register touch state callback
mt.on(Tinkerforge.BrickletMultiTouch.CALLBACK_TOUCH_STATE,
    // Callback function for touch state
    function(touchState) {
        var s = '';
        if(touchState & (1 << 12)) {
            s += 'In proximity, ';
        }
        if((touchState & 0xFFF) === 0) {
            s += 'No electrodes touched\n';
        }
        else {
            s += 'Electrodes ';
            for(var i=0; i<12; i++) {
                if(touchState & (1 << i)) {
                    s += parseInt(i) + ' ';
                }
            }
            s += 'touched\n';
        }
        console.log(s);
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/java/ExampleCallback.java
import com.tinkerforge.BrickletMultiTouch;
import com.tinkerforge.IPConnection;

public class ExampleCallback {
	private static final String HOST = "localhost";
	private static final int PORT = 4223;
	private static final String UID = "XYZ"; // Change to your UID
	
	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the documentation
	public static void main(String args[]) throws Exception {
		IPConnection ipcon = new IPConnection(); // Create IP connection
		BrickletMultiTouch mt = new BrickletMultiTouch(UID, ipcon); // Create device object

		ipcon.connect(HOST, PORT); // Connect to brickd
		// Don't use device before ipcon is connected

		// Add and implement touch state listener (called if touch state changes)
		mt.addTouchStateListener(new BrickletMultiTouch.TouchStateListener() {
			public void touchState(int touchState) {
				String str = "";

				if((touchState & (1 << 12)) == (1 << 12)) {
					str += "In proximity, ";
				}

				if((touchState & 0xfff) == 0) {
					str += "No electrodes touched" + System.getProperty("line.separator");
				} else {
					str += "Electrodes ";
					for(int i = 0; i < 12; i++) {
						if((touchState & (1 << i)) == (1 << i)) {
							str += i + " ";
						}
					}
					str += "touched" + System.getProperty("line.separator");
				}

				System.out.println(str);
			}
		});

		System.out.println("Press key to exit"); System.in.read();
		ipcon.disconnect();
	}
}
